### DISCUSSION TITLE
Buffer 中的 extrabuf 为什么  writable < extrabuf 就开辟两个内存块呢？

### DISCUSSION THREAD

**Discussion by CheNsiR-MASTER** [chenshuo/muduo#561] [General]
GitHub issues are for tracking bugs, not for general discussing like a forum.

If you have a general question to ask, send it to mailing list:

muduo-library@googlegroups.com

Or visit:
https://groups.google.com/forum/#!forum/muduo-library

When file an issue muduo, please provide a [SSCCE](http://sscce.org):
Short, Self Contained, Correct (Compilable), Example.

If you can't compile muduo, make sure you install `cmake` and `boost` from the
official package repository, e.g. `apt` or `yum`, before opening a bug.
Don't open a bug if you installed boost from a third-party source or
downloaded it by yourself, and couldn't compile muduo, thank you.

Also specify the exact environment where the issue occurs:

**Comment by CheNsiR-MASTER** [comment]
const int iovcnt = (writable < sizeof extrabuf) ? 2 : 1;

**Comment by chenshuo** [comment]
Right above this line:
```c
 // when there is enough space in this buffer, don't read into extrabuf.
 // when extrabuf is used, we read 128k-1 bytes at most.
 const int iovcnt = (writable < sizeof extrabuf) ? 2 : 1;
```
